### PR TITLE
[18938] Improve DX

### PR DESCRIPTION
**Pull request by peschina**
[comment]: <> (This file has been added on OGCIO fork)
### Ticket:

- [18938](https://dev.azure.com/OGCIO-Digital-Services/Digital%20Services%20Programme/_workitems/edit/18938)

### Description

Add a docker compose file that can be executed locally, with Logto, its database and our MyGovId mock service. The Logto image is pulled from dev ECR.
Remove redundant docker compose file used for db only.
Set mock endpoints in our local seeder file as environment variables, this gives us the flexibility needed to work with docker networking as well.
Add stages to the pipeline to build an image for our MyGovId mock service and push it to ECR only when merging into dev.

A follow-up PR will address the following points
- Seed mock endpoints using provided environment variables in docker compose config
- Pull MyGovId mock service mock service image from ECR
- Create a script to execute pre-requisite steps and run docker compose (aws and docker login, network creation etc)

## Type

- [ ] **Dependency upgrade**
- [ ] **Bug fix**
- [ ] **New feature**
- [x] **Dev change**

### Checklist:

- [ ] I have added tests that prove my fix is effective or that my feature works

### Screenshots:

N/A

### CODE REVIEW

**Comment by SamSalvatico** [comment]
Thank you @peschina 

This morning I was discussing about pulling images from ECR with @relvao.
The question is: how big would be the effort to onboard new developers on the project? 
@williamwd the ECR would be public or limited?

File: README.OGCIO.md [hunk 1]
```diff
@@ -60,12 +60,17 @@ PORT=3301
...
 USER_DEFAULT_ORGANIZATION_ROLE_NAMES=OGCIO Employee, OGCIO Manager
+MOCK_TOKEN_ENDPOINT=http://localhost:4005/logto/mock/token
+MOCK_KEYS_ENDPOINT=http://localhost:4005/logto/mock/keys
 ```
 2. Run the makefile command
 ```
 make run-native
 ```
 
+Hi guys, I was thinking about how to manage versioning of the Logto image we pull from ECR as part of the PR. Right now the image pushed is not versioned at all, each new push is just going to overwrite the exist image. I haven't talked about this William yet, but I think we want to keep things this way for simplicity (keep in mind this is also the image we use to deploy Logto in dev).
```
Comment (alfonsograziano) [inline README.OGCIO.md:71]: Shall this be here? it looks more like a message than a proper docs section

**Review by alfonsograziano — APPROVED** [review]
LGTM - even if we don't manage to have Logto on ECR, using it from local docker would still be a big improvement

File: README.OGCIO.md (hunk 1 above)
Comment (peschina) [reply README.OGCIO.md]: Nope, I must have pasted it by mistake :)

File: README.OGCIO.md [hunk 2]
```diff
@@ -38,11 +38,11 @@ You can also run Logto natively on your machine outside the docker container.
 
 If you start Logto natively, the database won't be available, and you will have to start it separately. The database is still dockerized and has its own Docker Compose configuration. Use the following command to start the database container:
 
-`docker compose -f docker-compose-db.yml up -d`
+`docker compose -f docker-compose-local.yml up -d postgres`
```
Comment (nnorbert) [inline README.OGCIO.md:41]: I'm not sure in this. We just need the DB to be up in Docker, the application should be started natively.

**Comment by SamSalvatico** [comment]
> LGTM - even if we don't manage to have Logto on ECR, using it from local docker would still be a big improvement

But remember that we have to test and fix seeded urls, because Docker container is not able to communicate with `localhost` when, for example, it tries to redirect to the callback url

**Review by nnorbert — APPROVED** [review]
LGTM

**Comment by peschina** [comment]
> But remember that we have to test and fix seeded urls, because Docker container is not able to communicate with localhost when, for example, it tries to redirect to the callback url

@SamSalvatico I've added environment variables for this, so when running with docker compose it will be able to resolve the communication service to service

**Pull request merged**